src/NoteDataUtil.cpp

```diff
@@ -961,11 +961,47 @@ struct recent_note_t
 		:row(r), track(t) {}
 };
 
+// CalculateRadarValues has to delay some stuff until a row ends, but can
+// only detect a row ending when it hits the next note.  There isn't a note
+// after the last row, so it also has to do the delayed stuff after exiting
+// its loop.  So this state structure exists to be passed to a function that
+// can be called from both places to do the work.  If this were Lua,
+// DoRowEndRadarCalc would be a nested function. -Kyz
+struct crv_state_t
```

Comment:
...joys of working at a place where three or more letter abbreviations are not encouraged unless we're talking standard abbreviations. Do whatever in this case.